build/appleRelease/scripts/downloadSignedMacOSBinaries.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+cliExecutableName=$1
+releaseVersion=$2
+goarch=$3
+GITHUB_ACCESS_TOKEN=$4
+
+# This script downloads signed macOS binaries for a specific version and architecture.
+
+# Function to retrieve the specific artifact URL with retries
+get_specific_artifact_url_with_retries() {
```

Comment:
Can we simply use curl with --retry, --retry-max-time and --retry-all-errors?
https://everything.curl.dev/usingcurl/downloads/retry.html